python_modules/libraries/dagster-airflow/dagster_airflow/dagster_job_factory.py

```diff
@@ -61,193 +82,140 @@ def make_dagster_job_from_airflow_dag(
-            database for each run. (default: False)
-        connections (List[Connection]): List of Airflow Connections to be created in the Ephemeral
-            Airflow DB, if use_emphemeral_airflow_db is False this will be ignored.
-
-    Returns:
-        Definitions
-    """
-    schedules, jobs = _make_schedules_and_jobs_from_airflow_dag_bag(
-        dag_bag,
-        use_airflow_template_context,
-        mock_xcom,
-        use_ephemeral_airflow_db,
-        connections,
-    )
-
-    return Definitions(
-        schedules=schedules,
-        jobs=jobs,
-    )
-
-
-def make_dagster_definitions_from_airflow_dags_path(
-    dag_path,
-    safe_mode=True,
-    use_airflow_template_context=False,
-    mock_xcom=False,
-    use_ephemeral_airflow_db=True,
-    connections=None,
-):
-    """Construct a Dagster repository corresponding to Airflow DAGs in dag_path.
-
-    Usage:
-        Create ``make_dagster_definitions.py``:
-
-        .. code-block:: python
-
-            from dagster_airflow import make_dagster_definitions_from_airflow_dags_path
+    check.inst_param(dag, "dag", DAG)
+    tags = check.opt_mapping_param(tags, "tags")
+    unique_id = check.opt_int_param(unique_id, "unique_id")
+    connections = check.opt_list_param(connections, "connections", of_type=Connection)
 
-            def make_definitions_from_dir():
-                return make_dagster_definitions_from_airflow_dags_path(
-                    '/path/to/dags/',
+    mutated_tags = dict(tags)
+    if IS_AIRFLOW_INGEST_PIPELINE_STR not in tags:
+        mutated_tags[IS_AIRFLOW_INGEST_PIPELINE_STR] = "true"
+
+    mutated_tags = validate_tags(mutated_tags)
+
+    node_dependencies, node_defs = get_graph_definition_args(dag=dag, unique_id=unique_id)
+
+    serialized_connections = serialize_connections(connections)
+
+    def airflow_db(context: InitResourceContext):
+        airflow_home_path = os.path.join(tempfile.gettempdir(), f"dagster_airflow_{context.run_id}")
+        os.environ["AIRFLOW_HOME"] = airflow_home_path
+        os.makedirs(airflow_home_path, exist_ok=True)
+        with Locker(airflow_home_path):
+            airflow_initialized = os.path.exists(f"{airflow_home_path}/airflow.db")
+            # because AIRFLOW_HOME has been overriden airflow needs to be reloaded
+            if is_airflow_2():
+                importlib.reload(airflow.configuration)
+                importlib.reload(airflow.settings)
+                importlib.reload(airflow)
+            else:
+                importlib.reload(airflow)
+            if not airflow_initialized:
+                db.initdb()
+                create_airflow_connections(
+                    [Connection(**c) for c in context.resource_config["connections"]]
+                )
+            dag = context.resources.dag
+            run_tags = context.dagster_run.tags if context.dagster_run else {}
+            if AIRFLOW_EXECUTION_DATE_STR in run_tags:
```

Comment:
i generally prefer to have this wrapped in a well-named helper rather than rely on comments